consensus/core/src/transaction.rs

```diff
@@ -37,6 +38,18 @@ pub(crate) struct TransactionConsumer {
     max_consumed_bytes_per_request: u64,
     max_consumed_transactions_per_request: u64,
     pending_transactions: Option<TransactionsGuard>,
+    block_status_subscribers: Arc<Mutex<BTreeMap<BlockRef, Vec<oneshot::Sender<BlockStatus>>>>>,
+}
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+#[allow(unused)]
+pub enum BlockStatus {
```

Comment:
tbh it is a bit confusing to me to have structs with the same name but different semantics in sui and consensus.

Reply:
Yeah I'll just re-use the consensus types. No reason to think of not "leaking" types from consensus to rest of codebase.